### ISSUE TITLE
checkhealth 'terminal' section should show detected terminal features like input mode

### ISSUE THREAD

**Issue by LunarLambda** [neovim/neovim#31113]
### Problem

There is a number of heuristics the Neovim TUI performs on startup to detect terminal features like CSI-u/Kitty protocol/modifyOtherKeys, truecolor/24-bit color support, and a few other modes/features.

Some examples from `terminfo_start` and friends:
<https://github.com/neovim/neovim/blob/master/src/nvim/tui/tui.c#L407>
<https://github.com/neovim/neovim/blob/master/src/nvim/tui/tui.c#L441>
<https://github.com/neovim/neovim/blob/master/src/nvim/tui/tui.c#L453>
<https://github.com/neovim/neovim/blob/master/src/nvim/tui/tui.c#L459>

For a lot of these, there isn't a good way to test that they are actually working, beyond things like 'try mapping `<C-S-Tab>` and see if it works' or 'try using a highlight group with extended underline styles'.

The fact the Neovim [FAQ](https://neovim.io/doc/user/faq.html) lists a large number of potential TUI issues indicates to me that some of this information/these checks could/should have a place in checkhealth.

### Expected behavior

It would be valuable troubleshooting information if the healthcheck could show information about what Neovim knows and expects about the terminal based on the information available to it.

I don't have a comprehensive list, but at minimum, I think it would be good to show the input mode (legacy/modifyOtherKeys/CSI u/kitty) and the detected terminal type (to see if e.g. the detection fell back to something generic like `xterm` when it shouldn't have), and ideally any other features that directly affect the functionality of the TUI but aren't exposed through options (e.g. extended underline support), though that's more for-the-users-interest than for debugging.

**Comment by LunarLambda** [comment]
Yeah, this intersects somewhat with the work that's been happening towards allowing TUI clients to be differentiated/detached and possibly having per-TUI settings for things like `termguicolors`.

I think the TUI client could 'just' report these things via set_client_info, which the TUI calls since https://github.com/neovim/neovim/pull/30397

Though this also would require some way to figure out 'which client did this command/input come from' so that checkhealth can retrieve the specific client's info, which is one of the TODOs in https://github.com/neovim/neovim/pull/30319

**Comment by justinmk** [comment]
Which of these are not indicated by `nvim -v3` (verbose=3)? 

> I think the TUI client could 'just' report these things via set_client_info, which the TUI calls since [#30397](https://github.com/neovim/neovim/pull/30397)

True, a programmatic interface might be useful.
